Models/AbstractProjectModels/AbstractUserModel.php

```diff
@@ -160,16 +184,345 @@ function (?string $value, string $key) {
     protected function passwordVerify(string $userPass, string $adminPass = null): bool
     {
         if (!password_verify($userPass, $this->getUser()['pass'])) {
-            $this->msgModel->setMsg($this->msgModel->getMessage('login', 'failed_pass'));
+            $this->msgModel->setMsg(
+                $this->msgModel->getMessage('failure', 'pass'),
+                'pass'
+            );
 
             return false;
         }
 
         return true;
     }
 
+    public function getUserSessionData(): ?array
+    {
+        return $this->sessionInfo->getUser();
+    }
+
     public function logout(): void
     {
         $this->sessionInfo->destroy();
     }
+
+    /**
+     * @return bool
+     * @throws \Exception
+     */
+    public function profile(): bool
+    {
+        $conditionData = $this->userIdInArrayFromSess();
+        $requestFields = [
+            'login',
+            'pass',
+            'name',
+            'birthdate',
+            'email',
+            'phone',
+            'address',
+            'image'
+        ];
+        if (!$this->userExist($this->getCustomerTable(), $requestFields, $conditionData)) {
+            $this->msgModel->errorMsgSetter(self::DEFAULT_ERROR);
+
+            return false;
+        }
+        $this->user['pass'] = 'Пароль не показывается в целях безопасности!';
+
+        return true;
+    }
+
+    /**
+     * @param array $fieldsArray
+     * @param array $dataArray
+     * @return array
+     * @throws \Exception
+     */
+    public function arrayDataCreator(array $fieldsArray, array $dataArray): array
+    {
+        $resultArray = [];
+        foreach ($fieldsArray as $field) {
+            if (array_key_exists($field, $dataArray)) {
+                $resultArray[$field] = $dataArray[$field];
+            } elseif ($field === 'phone' && !array_key_exists($field, $dataArray)) {
+                continue;
+            } else {
+                throw new \Exception('Field: \'' . $field . '\' doesn\'t exist!');
```

Comment:
You can use basic php  \InvalidArgumentException     

`            throw new \InvalidArgumentException('Field \'' . $field . '\' does not exist in the provided data array.');
`